### ISSUE TITLE
Daily stand up log

### ISSUE THREAD

**Issue by arrested-developer** [yalla-coop/rent-check#31]
Stand ups will happen every Wednesday and Thursday at 10am UK / 12pm Gaza time

**Comment by arrested-developer** [comment]
# Wednesday 18th September

## @thejoefriel 

1/2 day last week, finished wireframes and put into issues

- will work on tech spike for auth0

## @dupreesi 

restructured folders and refactored to serverless setup
1/2 day last week

- will start on admin panel / tech spike auth0

## @FarahZaqout

Auth0 does everything we want it to do, we can use social media for sign up and we can link it to our own database using the user object that is returned on sign-in.

Implemented in Node.js, so need to think about how to do it with serverless setup

2 days last week

## @amusameh 

Connecting database with netlify serverless functions, also had a look at how Auth0 works. Fixed eslint setup.

- will make a PR for eslint
- will share a branch

2 days last week

## @arrested-developer

demonstrated serverless functions
started looking into mongo setup
raised a PR with further clean-ups to the existing codebase

1 day last week

- will look at getting existing app migrated to mongodb

**Comment by arrested-developer** [comment]
## Thursday 19th Sep

## @dupreesi 
started work on admin panel, need another 2 hours or so
made a PR to add React Router
today will continue on admin panel components

## @thejoefriel

will review Simon's router PR
researched Auth0
merged in clean-up PR
started working on common components to be used across the app
finished menu
- mongo setup
- eslint

## @farahzaqout

have followed auth0 docs and made demo
not so happy with the code
will refactor for consistency
need to work on configuration

## @arrested-developer 

worked on understanding mongo
attended meeting to demo the app

**Comment by arrested-developer** [comment]
# Friday 20 september

## @thejoefriel 

worked on @amusameh 's PR for mongo, made some minor changes to DB connection which caused issues with tests
configured travis for mongo
added initial tests for db build
PR needs merge
**someone could pick up this branch to work on the protected route HOC**
1.5 days this week

## @dupreesi 

worked on admin panel interface
PR raised
new file structure
`/admin` route shows admin interface
some hardcoded data to show users and data sets
no frontend tests yet
PR needs review
2 days

## @farahzaqout

1 day so far this week, working today
- reconfiguration and refactoring
- looked into how to delete users - simple API call
will clean up the file structure and auth0, add documentation

## @arrested-developer 
1 day so far this week, working today
- will go over the backlog and try to make it easier to work from
- will look over PRs and comment / merge
- will work on private route HOC in @thejoefriel 's branch if there is time

**Comment by arrested-developer** [comment]
@FarahZaqout 

Auth0 - made a branch, worked on issues 1, 2, 3
Block: can't get environment vars working in the frontend, even with prefixing
Needs to assign himself to issues
Last week 2 days

@amusameh 

updated files with eslint and refactored
converted components to hooks
(PR raised for each)
Block: need PRs reviewed. ESlint PR needs merge first
Last week 2 days

@arrested-developer 
worked on @thejoefriel 's PR and added lambda function to get the map points from the database, not Airtable
Will look over @amusameh 's PRs and hooks documentation
A bit blocked waiting for auth PR before it's clear what to continue with

@dupreesi 

worked on admin panel components last week
will work on pulling users into the admin panel

@thejoefriel 

set up mongo and raised PR
today need to spend time looking at hooks and review the PR
common components PR hasn't been merged - blocked because waiting for auth PR
will refactor menu to use hooks
will work on pulling rental data into admin panel

**Comment by arrested-developer** [comment]
## Thursday 

@dupreesi 

raised PR for admin panel, now merged
will work on the issue for admin to grant and remove super user priveleges

Block: The handler functions work fine, but for tests they are not working. the `context.callbackWaitsForEmptyEventLoop = false;` doesn't seem to stop the function from timing out in tests.

@arrested-developer 

reviewed @amusameh and @dupreesi 's PRs
Made an addLocation lambda route
refactored lambda file structure

@thejoefriel 
1/2 day weds
working on issue to view all rental data (#14)
was blocked by lambda file structure but clear now

@amusameh 

working on the form to enter rental data - it took longer than expected because working on modularising 

@FarahZaqout 

Looked into how to set roles in Auth0 - looks simple, we need to write some config
Block: environment vars still not working in `components/index.js` - @amusameh suggested making sure it's prefixed in both places
Will raise PR and then work on adding user roles

## Testing handlers and queries

We're agreed to test both the queries and the handlers

**Comment by arrested-developer** [comment]
@dupreesi 
was working on the issue to allow admin to grant or remove super user privileges. Worked on the backend, query, tests and handler. Will continue, needs another hour or so.
Block: what to call patch hook - will work on this with high priority

@FarahZaqout 
Continued working on auth. With auth0 the token is not stored in the browser. Main block is difficulty of working with social logins e.g. google. We can remove that block as it's not in scope. Will continue working on auth, and should finish by end of today.

@thejoefriel 
Last week 1.5 days
Worked on #14 - rendering rental data in the admin panel. Wrote and tested queries. Realised that the issue also includes looking at the individual rental record, still working on this.
Will move onto verification / #16 (blocked by @dupreesi 's patch method hook)

@amusameh 
Working on #21 the form for rental data - built front end with validation and raised PR. PR is ready for review. Will move on to the same form for the admin.
Block: want to make sure the validation is accurate

@arrested-developer 
Last week 1.5 days
Worked on #60, #61, #62 
Reviewed @FarahZaqout 's auth PR - waiting for changes
Will make changes to #58 this morning, and then will not work today. Back tomorrow and Friday

**Comment by arrested-developer** [comment]
## Thursday 3rd October

@FarahZaqout 
Worked on the auth branch. Auth0 is unusual in that it doesn't seem to use a JWT when authenticating. Worked on getting a token.

@thejoefriel 
Didn't manage to work yesterday, will continue on the same issue

@amusameh 
Worked on the rental data form, will continue to work on the admin form for rental data

@dupreesi 
Worked on learning based on refactoring the useFetch hook yesterday. Will today make a PR which allows us to use GET, POST, PATCH methods.
possible block: will ask for support if neeed

@arrested-developer 
Wasn't working yesterday, but did review @amusameh 's PR and raise a new PR to clarify which fields are required. Today will continue to work on this and will start work on making the admin components mobile responsive.

**Comment by arrested-developer** [comment]
@FarahZaqout 

Played around with user roles - creating, removing, assigning. block as not sure what to do beyond that as the issue is vague (#9)

@arrested-developer 

Have been working on makings user responsive using a List. Have been blocked by new changes to the table being added. Have been reviewing PRs.
Want to work on getting the PRs merged and working out the order to merge them in

@amusameh 

Has been reviewing PRs and requesting changes. Will review Joe and Simon's PRs. His PRs need reviewing.

@dupreesi and @thejoefriel are travelling and will try to contribute later

---

Today's focus is to review and merge the outstanding PRs - @amusameh and @arrested-developer will review #59 and @FarahZaqout will start reviewing the others with focus on #75.

**Comment by arrested-developer** [comment]
## Thursday

@FarahZaqout 

pushed a refactor to the useFetch hook and reviewed @amusameh 's PRs. Makes sense for @amusameh to review the others.
Want to start working on #9 

@amusameh 

reviewed #75 and added some changes. Possible block - there will be some conflicts with #75 and #84.
Will review #75 and review farah's

@arrested-developer 

did refactoring on #75 - refactored
today will review farah's PR, work on getting PRs merged in, then continue working on refactoring the users table to a list.
Potential block - once working on refactoring the table, it will help for others to stop working on the Admin/users components

**Comment by arrested-developer** [comment]
@dupreesi 

No days last week. Will work this afternoon on merge conflicts in PR and then look at other PRs then issues

@arrested-developer 

Last week tried refactoring to Express instead of serverless functions. Want to be sure that we're authenticating securely and not opening up vulnerabilities. Also started to read about data ownership.

Will review auth PR and then other PRs, will write questions for Ivan to look over, and continue to read about data ownership.

@amusameh 

made changes to the dummy data PR as requested. Looking in to Leaflet performance with many data points as it makes it much slower.

block: when using auth0, user data is in auth0 so it's hard to have it in sync with each person's local db

@thejoefriel 

no days last week. will go through PR with Simon.

@FarahZaqout 

block: when starting the server locally it throws errors  (on master branch) - need to work out why. - could be that database needs to be rebuilt.

Worked on user access. Started looking into backend authentication.

**Comment by arrested-developer** [comment]
@arrested-developer 
express refactor ready for review, #87 / #88 
want to work on refactoring tables to lists but aware that we keep having conflicts so may be blocked

@thejoefriel and @dupreesi 
fixed merge conflicts from updated master branch, now fixing conflicts with this branch and express refactor.

@FarahZaqout 
has been preparing auth branch to work with Express will keep going with auth and prepare PR

@amusameh 
has been reviewing express refactor, will try to find something to work on but may be blocked until Joe and SImon's PR is ready

**Comment by dupreesi** [comment]
IMPORTANT: 
Team might need to setup joint discussion today as not having auth in place is currently a huge blocker.

@arrested-developer
Pushed PR but needs to pass first to be reviewed
Lets team know what he'll work on next

@thejoefriel and @dupreesi
Fixed yet another set of merge conflicts and finally raised PR #91. Needs Codecov sorting to pass which is of highest prio. 
Will then review @amusameh's PR #90 once it's been merged into latest master (including PR #91)
Will then pick up something new

@FarahZaqout
Will let team know how far he got with authentication work. 

@amusameh
Will review PR #91 once ready. 
Will then update PR #90.
Set up PR for Eslint fix wihich is ready to be reviewed
started #18(backend finished and waits for joe/simons pr to be merged) 
started #22

**Comment by arrested-developer** [comment]
@dupreesi did the auth0 nextJS tutorial - there is definitely a learning curve to nextjs. It seems like the same tech stack and possible to refactor. Couldn't find out if using JWT in cookie system is too easy to hack. Looked at the demo branch and it looks good but may take some time to refactor.

@thejoefriel did some research. It seems not too complicated. If everyone wants to use nextjs I think it's ok. If we refactor lets keep it a clean PR.

@amusameh didn't do the tutorial, went through docs on nextjs. Doesn't seem like a big change.

@arrested-developer did the auth0 nextjs tutorial, refactored a demo branch to show, which includes refactored tables for user, but this also removes any admin features that had been added

@FarahZaqout worked on auth with Express but getting CORS errors.

Agreed that we will try using nextjs as auth with Express has hit some roadblocks. @thejoefriel @amusameh and @FarahZaqout to do the auth0 nextjs tutorial, @dupreesi and @arrested-developer will start whiteboarding, making issues and begin the refactor. Each person to limit themselves to 1 full day this week.
